crates/payload/builder/src/traits.rs

```diff
@@ -56,10 +58,7 @@ pub trait PayloadJob: Future<Output = Result<(), PayloadBuilderError>> + Send +
     ///
     /// If `wait_for_pending` is provided and there's a payload building job in progress, returned
     /// future will first poll it until completion to get a potentially better payload.
-    fn resolve(
-        &mut self,
-        wait_for_pending: bool,
-    ) -> (Self::ResolvePayloadFuture, KeepPayloadJobAlive);
+    fn resolve(&mut self, kind: PayloadKind) -> (Self::ResolvePayloadFuture, KeepPayloadJobAlive);
```

Comment:
I want to keep this non breaking because this is currently used by users.

we should instead introduce a new function that accepts the kind and `resolve` calls this with `Kind::Earliest`